modules/effects/src/main/scala/dev/profunktor/redis4cats/extensions/luaScripting.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2018-2021 ProfunKtor
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.profunktor.redis4cats.extensions
+
+import cats.effect.kernel.{ Resource, Sync }
+import cats.syntax.all._
+import cats.{ Applicative, ApplicativeThrow }
+import dev.profunktor.redis4cats.algebra.Scripting
+import dev.profunktor.redis4cats.effects.ScriptOutputType
+import io.lettuce.core.RedisNoScriptException
+
+object luaScripting {
+
+  final case class LuaScript(contents: String, sha: String)
+
+  object LuaScript {
+
+    def make[F[_]: Applicative](redis: Scripting[F, _, _])(contents: String): F[LuaScript] =
+      redis.digest(contents).map(sha => LuaScript(contents, sha))
+
+    /** Helper to load a lua script from resources/lua/{resourceName}. The path to the lua scripts can be configured.
+      * @param redis
+      *   redis commands
+      * @param resourceName
+      *   filename of the lua script
+      * @param pathToScripts
+      *   path to the lua scripts
+      * @return
+      *   [[LuaScript]]
+      */
+    def loadFromResources[F[_]: Sync](redis: Scripting[F, _, _])(
+        resourceName: String,
+        pathToScripts: String = "lua"
+    ): F[LuaScript] =
+      Resource
+        .fromAutoCloseable(
+          Sync[F].delay(
+            scala.io.Source.fromResource(resource = s"$pathToScripts/$resourceName")
+          )
+        )
+        .evalMap { fileSrc =>
+          Sync[F].delay(fileSrc.getLines().mkString(""))
```

Comment:
We strip out the new lines from the file this way, no?

Do we need to use `blocking` instead of `delay` for these calls?

Reply:
> Do we need to use blocking instead of delay for these calls?

Great callout, thanks. Seems like all these File ops are blocking. Updated in latest commit.

> We strip out the new lines from the file this way, no?

Lua doesn't require newlines, however latest commit preserves them.